packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -1337,6 +1337,7 @@ function getReactiveHookCallbackIndex(calleeNode, options) {
   switch (node.name) {
     case 'useEffect':
     case 'useLayoutEffect':
+    case 'useHydrateableEffect':
```

Comment:
Not so sure about the naming. We don't really hydrate an effect since it those are never executed on the server anyway. I only used hydration in the name to communicate that the behavior of the hook is tied to hydration.